evalml/tests/automl_tests/test_automl.py

```diff
@@ -1221,8 +1233,12 @@ def fit(self, *args, **kwargs):
     mock_score_1 = MagicMock(return_value=mock_scores)
     DummyPipeline.score = mock_score_1
 
-    automl = AutoMLSearch(problem_type=problem_type, max_iterations=2,
-                          allowed_pipelines=[DummyPipeline], objective="auto")
+    if problem_type == "time series regression":
```

Comment:
I think we can eliminate the if/else here too. My concern is having to maintain the if/else when time series binary and multiclass get added. 

The good thing about the if is that it makes it clear in which cases the `problem_configuration` is used. Maybe we can add a comment to the test saying something like "Specifying problem_configuration for all problem types for conciseness"?